File: AppMapper/Profiles/InvoiceProfile.cs
using App.Contracts.Models.Output;
using AutoMapper;
using Domain.Contracts.Models;

namespace AppMapper.Profiles
{
    internal class InvoiceProfile: Profile
    {
        #region constructor

        public InvoiceProfile()
        {
            CreateMap<Invoice, InvoiceOutputModel>()
                .ForMember(dest => dest.Id, opt => opt.MapFrom(src => src.Id))
                .ForMember(dest => dest.Amount, opt => opt.MapFrom(src => src.Amount))
                .ForMember(dest => dest.Date, opt => opt.MapFrom(src => src.Date))
                .ForMember(dest => dest.StatusId, opt => opt.MapFrom(src => src.StatusId))
                .ForMember(dest => dest.Status, opt => opt.MapFrom(src => src.Status.Name));
        }

        #endregion
    }
}

File: UI/src/app/core/mappers/profiles/invoice.profile.ts
import { MappingProfile, createMap, forMember, mapFrom } from "@automapper/core";
import { InvoiceInputModel } from "../../models/input/invoice-input.model";
import { InvoiceModel } from "../../models/inner/invoice.model";

export const invoiceProfile: MappingProfile = (mapper) => {
    createMap(
        mapper,
        InvoiceInputModel,
        InvoiceModel,
        forMember(
            (dest) => dest.id,
            mapFrom((source) => source.id)
        ),
        forMember(
            (dest) => dest.amount,
            mapFrom((source) => source.amount)
        ),
        forMember(
            (dest) => dest.statusId,
            mapFrom((source) => source.statusId)
        ),
        forMember(
            (dest) => dest.status,
            mapFrom((source) => source.status)
        ),
        forMember(
            (dest) => dest.date,
            mapFrom((source) => new Date(source.date))
        ),
    );
};